src/errors/RequestResponseError.ts

```diff
@@ -64,7 +64,7 @@ const serializeErrorResponse = ({ code, httpCode, errors }: FormattedErrorReques
     return `Code: ${code}, HttpCode: ${httpCode}, Errors: ${formattedErrors.join(", ")}`;
 };
 
-export const fetchErrorReasons = (errorRequest: RawErrorRequest) => {
+export const fetchErrorReasons = (errorRequest: ErrorResponse) => {
```

Comment:
@phieronymus 
Just to confirm, does the api ever return more than one error? If not, I think there is no need to return the reason as Array.

Reply:
It is always one error response that can contain several errors in the `errors` array 🙂

Reply:
Okay. What I think is the error what is linked to the field is only one and we may want to specify the error. Which is good to add the logic, the SDK or common?

Reply:
Sorry, I did not understand the previous message. Do you have an example?

Reply:
Sure. I am concerned about the following case.
```
{
    code: "VALIDATION_ERROR",
    status: "error",
    errors: [
        { field: "first-field", reason: "NUMBER_MIN" },
        { field: "second-field", reason: "CANNOT_BE_ZERO" },
    ]
}
```
On the current implementation, when `fetchErrorReasons` receive the above values, it returns `["NUMBER_MIN", "CANNOT_BE_ZERO"]`
But if it return these values, I am thinking that it is going to be hard to tell which error is from which field.
If the error object has always only one pair of field and reason, there are no problem, but actually it can have several pairs, so I think we need a logic to specify them.
In this case I also think we need to the way how to link the field.

Sorry if this is outlandish.

Reply:
We will need to keep this format though (the pair with the field). You can then link them to the fields using the `mapAPIReponseToError` in the `Details` and `Create` components.

Reply:
Okay! Thanks :bow: